### ISSUE TITLE
Could not build module 'Mapbox' (MGLAccountManager.h file not found)

### ISSUE THREAD

**Issue by munkius** [nitaliano/react-native-mapbox-gl#270]
I followed the steps in the installation guide file, but I get the error the Mapbox framework could not be found. It get this after adding `libRCTMapboxGL.a` to the section `Link Binaries With Libraries`. Before doing this I can successfully build my project.

I'm working with react-native-mapbox-gl version 3.2.1.

![image](https://cloud.githubusercontent.com/assets/194593/12921947/d06eac36-cf4f-11e5-894c-edb959d82f2f.png)

**Comment by JonnyBGod** [comment]
Same Error

**Comment by bsudekum** [comment]
Could you verify:
1. `Mapbox.framework` is listed in `Embedded Binaries`
2. `Linked frameworks and Libraries` has `libRCTMapboxGL.a`

![](https://cldup.com/cOZRZt2jZG.png)

Uploading a screenshot just like this would help

**Comment by ferndopolis** [comment]
Same error on 3.2.1. When I search for `MGLAccountManaer.h` in my project root

```
$ find . -type f -name "MGLAccountManager.h"                                                             
 ./node_modules/react-native-mapbox-gl/ios/Mapbox.framework/Mapbox.framework/Headers/MGLAccountManager.h
```

there is a Mapbox.framework nested inside Mapbox.framework.  Is that right?

**Comment by munkius** [comment]
![image](https://cloud.githubusercontent.com/assets/194593/12984447/0e63a0b0-d0ee-11e5-9a1d-fed896b04c20.png)

**Comment by bsudekum** [comment]
Hmmm that looks like. @ferndopolis something is up with your install process, when I run 

``` bash
find . -type f -name "MGLAccountManager.h"  
./node_modules/react-native-mapbox-gl/ios/Mapbox.framework/Headers/MGLAccountManager.h
```

Unsure how it would put the framework inside of the framework...

@munkius are you upgrading from a previous version?

**Comment by munkius** [comment]
@bsudekum It's a clean install, what makes you think that?

**Comment by bsudekum** [comment]
@munkius just wondering. An older version could have been messing with the build.

**Comment by munkius** [comment]
Tried it a bunch of times with a fres workspace to be sure not to disturb you guys with a nonsense issue :) Hope this can work out, I'm really curious about getting Mapbox to work within react native.

**Comment by bsudekum** [comment]
@munkius Here is a barebones setup: https://cloudup.com/c9J25rX9Jla

Can you download it, unzip it and try and run it? Maybe see if anything is different.

**Comment by munkius** [comment]
That builds just fine :) Plus, when I copy the `node_modules/react-native-mapbox-gl` folder towards my test project (after removing the previous version), that project builds successfully as well.

**Comment by jordangarside** [comment]
Getting the same issue in my project.
RN 0.19
MapboxGL 3.21

**Comment by jordangarside** [comment]
Got it to work, did `react-native upgrade` to reset the project.
Then I made sure to `copy files if necessary` for both the .xcode import and the .framework import.
I also had to make sure to set target to iOS version 9 instead of 7 which was the default for the RN project (MapboxGL seems to target 8).
Might want to make that more clear in the instructions.

Thanks!

**Comment by aksonov** [comment]
I have the same issue. It is very weird, i see react-native-mapbox-gl/ios/Mapbox.framework _without_ MGLAccountManager.h and react-native-mapbox-gl/ios/Mapbox.framework/Mapbox.framework _with_ that file...

**Comment by samuelngs** [comment]
same issue here. It was working last week and it suddently breaks after I reinstall npm packages with `npm install`

**Comment by bsudekum** [comment]
@samuelngs @aksonov the install process [changed in 3.2.0](https://github.com/mapbox/react-native-mapbox-gl/blob/master/CHANGELOG.md#v320). Have you gone through the new install process?

**Comment by aksonov** [comment]
@bsudekum I did all these steps several times without success.

**Comment by aksonov** [comment]
setting target to 8.0 or 9.0 also didn't solve the problem. Why there are nested Mapbox.framework files?

**Comment by bsudekum** [comment]
@aksonov I'm not seeing nested frameworks when doing a fresh install. Here is what my completed install process looks like: https://cloudup.com/c9J25rX9Jla

**Comment by samuelngs** [comment]
@bsudekum I upgraded the mapboxgl to v3.2.0 last week and it worked fine before, yeah and I had gone through the new install process (the one with mapbox.framework).

**Comment by jordangarside** [comment]
I second @munkius and the copying of the `react-native-mapbox-gl` folder from @bsudekum worked for me
Edit: See the next posts before trying this, left for reference.

**Comment by bsudekum** [comment]
I'm confused as to how these folders would look any different.

**Comment by jordangarside** [comment]
@bsudekum
So it turns out that wasn't my problem, the problem is that 3.2.3 isn't working for me now (same issue as this one).
I tried to get 3.2.3 to work quite a few times on my project and also by starting a new one (with react-native init) and neither worked.
Switching to the 3.2.2 package worked as expected (that's the one you have in the working example).

It's really important that anyone doing this selects `Copy files if necessary` on both the .xcode import and the .framework import. That used to be in the old instructions.

Still not sure how to get 3.2.3 to work, but I give up for now.

**Comment by aksonov** [comment]
@bsudekum It uses 3.2.2, not 3.2.3. After i've removed node_modules and run npm install, i've got exactly the same issue....

**Comment by jarredwitt** [comment]
I was running into this issue as well. 

Mapbox 3.2.3
RN 0.19
Node 5.3.0
NPM 3.3.12

I was seeing the same issue as the others with Mapbox.framework containing a Mapbox.framework. I was able to solve the issue and get it to build by moving the nested Mapbox.framework to another location and then deleting the Mapbox.framework that was left. I then moved the nested Mapbox.framework back into the ios directory within my node_modules/react-native-mapbox-gl and the build was able to succeed. Hope this helps. Let me know I need to clarify anything else.
